### PR TITLE
MRG: Add fixed dipole pos mode

### PR DESCRIPTION
**Pull request by Eric89GXL**
This is a feature of Xfit that would be nice to have -- constrain `pos` and `ori` when doing dipole fitting so time-courses can be obtained.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Changing to WIP because I need to check to see the correspondence now between mne-python's data and Xfit's (argh)

**Comment by Eric89GXL** [comment]
Okay got a hold of some more files, I think this solution is okay. Ready for review/merge from my end.

**Comment by Eric89GXL** [comment]
@agramfort rebased due to merge conflicts caused by recent merges, any time to review?

File: tutorials/plot_dipole_fit.py [hunk 1]
```diff
@@ -39,11 +39,19 @@
...
 
 # Fit a dipole
 dip = mne.fit_dipole(evoked, fname_cov, fname_bem, fname_trans)[0]
 
+# Estimate the time course of a single dipole (the one that maximized GOF)
+# over the entire interval
+best_idx = np.argmax(dip.gof)
+dip_fixed = mne.fit_dipole(evoked_full, fname_cov, fname_bem, fname_trans,
+                           pos=dip.pos[best_idx], ori=dip.ori[best_idx])[0]
+dip_fixed.plot()
```
Comment (agramfort) [inline tutorials/plot_dipole_fit.py]: this should be moved at the end after the version line of ####
to have cell separation in the rendering

Comment (agramfort) [inline tutorials/plot_dipole_fit.py]: also if you want to add some narrative please do. We have no excuse now :)

File: tutorials/plot_dipole_fit.py [hunk 2]
```diff
@@ -74,3 +76,10 @@
 diff.plot_topomap(time_format='Difference', axes=axes[2], **plot_params)
 plt.suptitle('Comparison of measured and predicted fields '
              'at {:.0f} ms'.format(best_time * 1000.), fontsize=16)
+
+###############################################################################
+# Estimate the time course of a single dipole with fixed position and
+# orientation (the one that maximized GOF)over the entire interval
+dip_fixed = mne.fit_dipole(evoked_full, fname_cov, fname_bem, fname_trans,
+                           pos=dip.pos[best_idx], ori=dip.ori[best_idx])[0]
+dip_fixed.plot()
```
Comment (Eric89GXL) [inline tutorials/plot_dipole_fit.py]: @agramfort better?

**Comment by agramfort** [comment]
yes. Merge when travis is happy.

**Comment by agramfort** [comment]
don't forget to update what's new

**Pull request merged**